frontend/src/forms/manage-restaurant-form/ManageRestaurantForm.tsx

```diff
@@ -46,17 +47,19 @@ const formSchema = z.object({
     })
   ),
 
+  imageUrl: z.string().optional(),
   imageFile: z.instanceof(File, { message: 'Image is required' }),
 });
 
 type RestaurantFormData = z.infer<typeof formSchema>;
 
 type Props = {
+  restaurant?: Restaurant;
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/33#issuecomment-2394670085>#d60155</a><span> - 10/04/2024, 09:21 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The form schema and initial form values are not updated to handle both creation and editing scenarios after adding the 'restaurant' prop.
</div>
<div id="fix">
<b>Fix:</b> Update the form schema and initial form values to handle both creation and editing scenarios. Modify the useEffect hook to properly initialize the form with existing restaurant data when the 'restaurant' prop is provided.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -63,9 +63,20 @@
    const form = useForm<RestaurantFormData>({
      resolver: zodResolver(formSchema),
      defaultValues: {
 +      restaurantName: '',
 +      city: '',
 +      country: '',
 +      deliveryPrice: 0,
 +      estimatedDeliveryTime: 0,
        cuisines: [],
 -      menuItems: [{ name: '', price: 1 }],
 +      menuItems: [],
 +      imageUrl: '',
      },
    });
 
    useEffect(() => {
      if (!restaurant) {
        return;
      }
 
      form.reset({
        restaurantName: restaurant.restaurantName,
        city: restaurant.city,
        country: restaurant.country,
        deliveryPrice: restaurant.deliveryPrice,
        estimatedDeliveryTime: restaurant.estimatedDeliveryTime,
        cuisines: restaurant.cuisines,
 -      menuItems: restaurant.menuItems,
 +      menuItems: restaurant.menuItems.length > 0 ? restaurant.menuItems : [{ name: '', price: 1 }],
 +      imageUrl: restaurant.imageUrl,
      });
    }, [restaurant, form]);
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged